k8s/src/main/scala/io/buoyant/k8s/ClientConfig.scala

```diff
@@ -0,0 +1,48 @@
+package io.buoyant.k8s
+
+import com.twitter.finagle.{Filter, Http, Stack}
+import com.twitter.finagle.http.{Request, Response}
+import io.buoyant.k8s
+import scala.io.Source
+
+/**
+ * Meant to be implemented by configuration classes that need to
+ * produce a k8s API client.
+ */
+trait ClientConfig {
+  def host: Option[String]
+  def portNum: Option[Int]
+  def tls: Option[Boolean]
+  def tlsWithoutValidation: Option[Boolean]
+  def authTokenFile: Option[String]
+
+  protected def getHost = host.getOrElse(Defaults.Host)
+
+  protected def getPort = portNum.getOrElse(Defaults.Port)
+
+  protected def dst = s"/$$/inet/$getHost/$getPort"
+
+  private def authFilter = authTokenFile match {
+    case Some(path) =>
+      val token = Source.fromFile(path).mkString
+      if (token.nonEmpty) new AuthFilter(token)
+      else Filter.identity[Request, Response]
+    case None => Filter.identity[Request, Response]
+  }
+
+  protected def mkClient(
+    params: Stack.Params = Stack.Params.empty
+  ) = {
+    val setHost = new SetHostFilter(getHost, getPort)
+    val client = (tls, tlsWithoutValidation) match {
+      case (Some(false), _) => Http.client
```

Comment:
Should the `(Some(false), Some(_))` case emit some kind of error or warning since it's not really valid?  ie specifying `tlsWithoutValidation` doesn't have meaning if `tls` is off